tests/io/vtk_writer_test.cc

```diff
@@ -4,16 +4,49 @@
 
 #include "vtk_writer.h"
 
-// Check ReadMeshAscii
+// Check Write VTK
 TEST_CASE("VTK Writer is checked", "[vtk][writer]") {
 
+  SECTION("Check scalar/vector/tensor file ") {
+    std::vector<Eigen::Matrix<double, 3, 1>> coordinates;
+    // VTK PolyData writer
+    auto vtk_writer = std::make_unique<VtkWriter>(coordinates);
+
+    SECTION("Write scalar data") {
+      const std::string vtk_file = "scalar_vtk.vtp";
+      const std::string attribute = "mass";
```

Comment:
Can you really output mass as the example? I see that `write_scalar_point_data` only allow output of variables inside the `state_vars` and mass or volume or material id, for instance, are not there.
https://github.com/cb-geo/mpm/blob/72cd3842961f89962ac66b87b823c5674f9ce2f9/include/solvers/mpm_base.tcc#L526-L527

Reply:
Will do "epds"

Reply:
How about all the other parameters outside `state_vars`? Any thoughts, cause I always hardcoded `particle.tcc` to output that, we might need to think about it...

Reply:
This function is purely for `state_vars`, if we need others we'd have to do something similar like we have in `vector_data`. Probably a different PR

Reply:
I see... That's okay. In that case, then all the particle private variables should be also included there too perhaps? Let's discuss this further.